racket/collects/net/http-client.rkt

```diff
@@ -121,34 +129,37 @@
                          #:headers [headers-bs empty]
                          #:content-decode [decodes '(gzip)]
                          #:data [data #f])
+  (define (print-to fmt . args)
+    ;; (eprintf "print-to: ~a~%" (apply format fmt args))
+    (apply fprintf to fmt args))
```

Comment:
Variable arguments and `apply` cause a lot of allocation and uncertainty. A macro version wouldn't have this problem if you wanted to do it that way.